### PR TITLE
fix: add proxy env propagation

### LINKED ISSUE #30165: Handle proxy related env vars in the Operator
### Description

The Operator should correctly handle proxy related env vars: `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY`. Those env vars are de facto a [standard](https://sdk.operatorframework.io/docs/building-operators/golang/references/proxy-vars/).

The Operator should take one of the following approaches:
* Ignore the env vars as it might not make much sense for the Operator to use proxy. Only outgoing requests are toward the Kube API.
* Properly support the env vars and use the specified proxy setting for all potential outgoing requests (incl. Kube API).
  * A best practice is also considered for the Operator to pass down the env vars to the Operand pods.

### Discussion

_No response_

### Motivation

Currently, it looks like the env vars are recognized by Fabric8, however the implementation might suffer from some issues. As reported for RHBK, an exception is thrown when the proxy env vars do not include a port:
```
ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
    at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
    at io.quarkus.runtime.Application.start(Application.java:101)
    at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.IllegalArgumentException: Failure in creating proxy URL. Proxy port is required!
    at io.fabric8.kubernetes.client.utils.HttpClientUtils.getProxyUri(HttpClientUtils.java:91)
    at io.fabric8.kubernetes.client.utils.HttpClientUtils.configureProxy(HttpClientUtils.java:210)
```
This is essentially a bug but since the Operator doesn't claim support for the proxy env vars, tracking this as an enhancement.

This behaviour is particularly problematic when a cluster-wide proxy is configured and the env vars implicitly set on the pods. In this case it might be problematic for the user to overwrite the env vars.

### Details

Seems to be related to: https://github.com/fabric8io/kubernetes-client/issues/3671

### PR DESCRIPTION
**Pull request by shawkins**
closes: #30165

This seems straight-forward enough, but if there are any concerns about tests then it could be based upon the quarkus config instead.

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java [hunk 1]
```diff
@@ -419,7 +422,14 @@ private List<EnvVar> getDefaultAndAdditionalEnvVars(Keycloak keycloakCR, String
                     }
                     return envBuilder.build();
                 })
-                .collect(Collectors.toList());
+                .collect(Collectors.toCollection(ArrayList::new));
+
+        for (String env : COPY_ENV) {
+            String value = System.getenv(env);
+            if (value != null) {
+                envVars.add(new EnvVar(env, value, null));
```
Comment (mabartos) [inline operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java:430]:
```suggestion
                envVars.add(new EnvVarBuilder().withName(env).withValue(value).build());
```

nitpick: I'd prefer to use the builder for it, as I can imagine some default `EnvVarSource` in the future. But it is only a nitpick, not a big deal.

**Review by mabartos — COMMENTED** [review]
@shawkins Changes almost LGTM, but I'd appreciate some simple test case even for such a straight-forward approach. WDYT?

I also retried the failed GHA to see if they were really caused by this PR.

**Comment by shawkins** [comment]
> @shawkins Changes almost LGTM, but I'd appreciate some simple test case even for such a straight-forward approach. WDYT?

Not sure there is a simple test case for this. Setting environment variables would generally need to be a container test. Using the quarkus config helps, but opens this up to other config sources. The last options would be static mocking or creating a bean for the env lookup so that it could be replaced more easily.

> I also retried the failed GHA to see if they were really caused by this PR.

Seems to have passed now. That was surprising.

**Comment by mabartos** [comment]
> Not sure there is a simple test case for this. Setting environment variables would generally need to be a container test. Using the quarkus config helps, but opens this up to other config sources. The last options would be static mocking or creating a bean for the env lookup so that it could be replaced more easily.

@shawkins Yep, I've just been thinking about the test case, and if it's something that would be more complicated, it probably is not worth the effort for simple changes like that. So, I'm ok to leave it as is.

**Review by mabartos — DISMISSED** [review]
@shawkins As mentioned, LGTM from my side.

However, it'd be good to have it approved by @vmuzikar as the author of the issue.

**Review by mabartos — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
@shawkins Thank you for the PR.

I agree it's quite straightforward but not really easy to test (we'd have to create remote-only test). LGTM as is.

**Pull request merged**